### PR TITLE
GitHub Actions: skip some jobs for forks/Dependabot

### LINKED ISSUE #6770: GitHub Actions: fix permissions for forks and Dependabot
## Task Description

Two recent significant security changes require us to rewrite the GitHub Actions so that we can safely accept code from Dependabot and other third party forks.

https://github.blog/changelog/2021-02-19-github-actions-workflows-triggered-by-dependabot-prs-will-run-with-read-only-permissions/

https://securitylab.github.com/research/github-actions-preventing-pwn-requests

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Skips some jobs and steps for PRs from forks as well as from Dependabot, as these don't have access to secrets and cannot do things like uploading snapshots or annotating PRs. 

Improves experience for outside contributors forking the project.

## Relevant Technical Choices

Updating GHA workflow files. Inspired by https://github.com/ampproject/amp-wp

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6770

### CODE REVIEW

File: .github/workflows/tests-e2e.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: '7.4'
-          coverage: none
+          snapshots: none
```
Comment (spacedmonkey) [inline .github/workflows/tests-e2e.yml]: Why change this?

**Review by spacedmonkey — APPROVED** [review]
SGTM

File: .github/workflows/tests-e2e.yml (hunk 1 above)
Comment (swissspidy) [reply .github/workflows/tests-e2e.yml]: Whoops, search/replace mistake. This should of course not be changed here.

File: .github/workflows/tests-e2e.yml [hunk 2]
```diff
@@ -41,10 +41,10 @@ jobs:
         include:
           - browser: 'chrome'
             wp: 'latest'
-            coverage: true
+            snapshots: true
```
Comment (spacedmonkey) [inline .github/workflows/tests-e2e.yml:44]: @swissspidy Revert this change here

File: .github/workflows/tests-e2e.yml [hunk 3]
```diff
@@ -41,10 +41,10 @@ jobs:
...
           - browser: 'chrome'
             wp: '5.3.6'
-            coverage: false
+            snapshots: false
```
Comment (spacedmonkey) [inline .github/workflows/tests-e2e.yml:47]: And here.

File: .github/workflows/tests-e2e.yml [hunk 4]
```diff
@@ -128,11 +128,11 @@ jobs:
         run: npm run test:e2e:percy
         env:
           PERCY_TOKEN: ${{ secrets.PERCY_TOKEN_E2E }}
-        if: ( matrix.coverage ) && ( github.event.pull_request.draft == false )
+        if: ( matrix.snapshots ) && ( github.event.pull_request.draft == false )
```
Comment (spacedmonkey) [inline .github/workflows/tests-e2e.yml:131]: And here.

File: .github/workflows/tests-e2e.yml [hunk 5]
```diff
@@ -128,11 +128,11 @@ jobs:
...
 
       - name: Run E2E tests
         run: npm run test:e2e
-        if: ( ! matrix.coverage ) || ( github.event.pull_request.draft == true )
+        if: ( ! matrix.snapshots ) || ( github.event.pull_request.draft == true )
```
Comment (spacedmonkey) [inline .github/workflows/tests-e2e.yml:135]: And here.

File: .github/workflows/tests-e2e.yml (hunk 2 above)
Comment (swissspidy) [reply .github/workflows/tests-e2e.yml:44]: Context:

I was looking at this workflow because I thought I'd needed it to change as well to prevent errors (turns out that wasn't needed).

When doing that, https://github.com/google/web-stories-wp/pull/5423#discussion_r530529914 came to mind.

IMHO here `snapshots` is clearer as it's about creating Percy snapshots, not about code coverage or anything.
